app/crud/crud_pest_model.py

```diff
@@ -0,0 +1,57 @@
+from typing import Optional
+
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from crud.base import CRUDBase
+
+from models import PestModel, Cultivation
+from schemas import CreatePestModel, UpdatePestModel
+
+
+class CrudPestModel(CRUDBase[PestModel, CreatePestModel, UpdatePestModel]):
+
+    def create(self, db: Session, obj_in: CreatePestModel, **kwargs) -> Optional[PestModel]:
+
+        pest_model_db = PestModel(name=obj_in.name, description=obj_in.description, geo_areas_of_application=obj_in.geo_areas_of_application)
+
+        # Attempt to create a pest model entity
+        db.add(pest_model_db)
+        try:
+            db.commit()
+        except SQLAlchemyError:
+            db.rollback()
+            return None
+        db.refresh(pest_model_db)
+
+        # Assemble the cultivations from the list
+        cultivations = [Cultivation(name=x, pest_model_id=pest_model_db.id) for x in obj_in.cultivations]
```

Comment:
I don't really understad. The cultivation (parcel) information are provided together with the pest_model information?

What will happen in the case that 2 pest_models refer to the same cultivation?

Reply:
The cultivation isn't parcel information, its a list of plant types: [grapevine, berries, ...] that this pest can attack/infest.
for example the Plasmopara viticola is a grapevine disease that doesn't infest other plants (as far as I understand it) like tubers or berry bushes, it only(?) attacks grapevines

Reply:
hmm i am only thinking if we need to store it in separate object.

Reply:
This is another one where I thought it was better to leave it open to expansion.
As it stands now it's a bit overkill, but I'm not sure if it will change (more fields) in the future?